packages/node/src/indexer/ds-processor.service.ts

```diff
@@ -84,18 +56,19 @@ export class DsProcessorService {
   getDsProcessor<D extends string, T extends SubqlNetworkFilter>(
```

Comment:
I'm not sure if we should be checking that the ds is part of the project data sources from the constructor

Reply:
Don't get the full context of your question. If the check is async, we can't do it in constructor though.

Reply:
Its possible to do something like this:

```ts
const project = {
  ...rest,
  dataSources: [ds1, ds2],
}

const ds3 = ...;

new DsProcessorService(project).getDsProcessor(ds3);
```

ds3 has nothing to do with the project